File: solution/3700-3799/3713.Longest Balanced Substring I/Solution.go
func longestBalanced(s string) (ans int) {
	n := len(s)
	for i := 0; i < n; i++ {
		cnt := [26]int{}
		mx, v := 0, 0
		for j := i; j < n; j++ {
			c := s[j] - 'a'
			cnt[c]++
			if cnt[c] == 1 {
				v++
			}
			mx = max(mx, cnt[c])
			if mx*v == j-i+1 {
				ans = max(ans, j-i+1)
			}
		}
	}

	return ans
}

File: solution/3700-3799/3713.Longest Balanced Substring I/Solution.cpp
class Solution {
public:
    int longestBalanced(string s) {
        int n = s.size();
        vector<int> cnt(26, 0);
        int ans = 0;
        for (int i = 0; i < n; ++i) {
            fill(cnt.begin(), cnt.end(), 0);
            int mx = 0, v = 0;
            for (int j = i; j < n; ++j) {
                int c = s[j] - 'a';
                if (++cnt[c] == 1) {
                    ++v;
                }
                mx = max(mx, cnt[c]);
                if (mx * v == j - i + 1) {
                    ans = max(ans, j - i + 1);
                }
            }
        }
        return ans;
    }
};
